File: Java/IntegerListFromUsp/src/Exercicio13.java
import java.util.Scanner;

public class Exercicio13 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        scanner.reset();
        System.out.print("Digite um número: ");
        int numero = scanner.nextInt();

        if (numero > 0) {
            int i = 1;
            int soma = 0;
            while (i < numero) {
                if (numero % i == 0) {
                    soma += i;
                }
                i++;
            }
            if (soma == numero) {
                System.out.println("%d é um número perfeito.".formatted(numero));
            } else {
                System.out.println("%d não é um número perfeito.".formatted(numero));
            }
        } else {
            System.out.println("Número digitado é menor que zero");
        }
    }

}

File: CPP/IntegerListFromUsp/src/Exercicio13.cpp
#include <iostream>

using namespace std;

int main() {
    int n;
    cout << "Digite um inteiro positivo n: ";
    cin >> n;

    int soma = 0;
    for (int divisor = 1; divisor < n; divisor++) {
        if (n % divisor == 0) {
            soma += divisor;
        }
    }

    cout << n << (soma == n ? " é perfeito." : " não é perfeito.") << endl;
    return 0;
}
